apps/gallery/test/marionette/lib/system.js

```diff
@@ -25,6 +26,13 @@ System.prototype = {
     this.client.apps.switchToApp(appOrigin);
   },
 
+  waitForActionMenuToDisplay: function() {
+    this.client.waitFor(function() {
+      return this.client.helper.waitForElement(
+        System.Selector.actionMenu).displayed();
+    }.bind(this));
+  },
+
```

Comment:
While running Gij test on my local, I saw pickand open image activity test intermittently fail  at the call
system.menuOptionButton('Gallery').tap();

On debugging, It looks like this.actionMenu waits for form[data-z-index-level="action-menu"] but exits before 'Gallery' option button is displayed.  Adding explicit waitForActionMenuToDisplay function helps to resolve this failure. I can try to change it to explicit wait for 'button' to display, I guess that will be more precise resolution of this race condition.